src/IB/IIMethod.cpp

```diff
@@ -1317,21 +1317,48 @@ IIMethod::interpolateVelocity(const int u_data_idx,
                             for (BoxIterator<NDIM> b(stencil_box); b; b++)
                             {
                                 const hier::Index<NDIM>& ic = b();
-                                for (int j = 0; j < NDIM; ++j) du_jump(j) = DU_jump_qp[d][s * NDIM + j];
+                                for (int j = 0; j < NDIM; ++j)
+                                {
+                                    du_jump(j) = DU_jump_qp[d][s * NDIM + j];
+                                    wrc(j) = wr[j][ic_upper[j] - ic[j]];
+                                }
 #if (NDIM == 2)
-                                coeff_vec =
-                                    VectorValue<double>(interpCoeff[ic[0]][ic[1]][0], interpCoeff[ic[0]][ic[1]][1]);
-                                Ujump[ic[0]][ic[1]][d] = dx[0] * w[0][ic[0] - ic_lower[0]] * w[1][ic[1] - ic_lower[1]] *
-                                                         (coeff_vec * du_jump);
+                                if (d_use_Tan_velocity_correction)
+                                {
+                                    // Use velocity correction term from
+                                    // https://epubs.siam.org/doi/abs/10.1137/080712970?casa_token=LevsVOzF2m8AAAAA:UAgURw3W4dny0SrV2Vv30tt8-tOtlhsuzCCLfair1jLPyIqlL0ob-qnFaaL7oaweopiA2lDOXKo
```

Comment:
```suggestion
                                    // https://epubs.siam.org/doi/abs/10.1137/080712970?
```
Please also add some information in a comment about how the two approaches differ from each other.

Reply:
Hello Boyce, I am not sure what comment I should include in the code. Could you please give me some suggestions? I could write down my understanding of difference here:
The difference between those two schemes is just the correction terms in velocity interpolation operator:
See Eqn. (3.12)-(3.15) in Sec. 3.3 in
https://epubs.siam.org/doi/epdf/10.1137/080712970
See Eqn. (27)-(28) in Sec. 3.5 in 
https://arxiv.org/pdf/1812.06840
The first version Tan derived base on generalized taylor expansiion. And it is second order according to the author and this is verified by Me and Michael's code.
I don't fully understand Amin's interpolation scheme (the one currently in the IBAMR). But according to what Amin wrote in the paper:
"This modified bilinear interpolation is similar to the approach of Tan et al. [72, 73]. An important difference
is in the value used for the distance from the Cartesian grid locations to the interface in the Taylor series
expansion of the correction terms. The approach by Tan et al. uses the distance between the grid points
and the interfacial point (|r| in Fig. 5). Here, this is modified by projecting the location vector r to the
normal vector n and using |r · n| instead. Taking into account for the normal vector naturally provides a
more accurate approximation to the actual distance between the grid point and the interface"

Reply:
Can you check to see if the current scheme gives the right answer for a piecewise linear field with a derivative discontinuity along the interface?

Reply:
Sure!

Reply:
Please make sure to test multiple interface positions and orientations.

Reply:
Hello Boyce, 
I constructed an interface that not parallel to either axis. Then I constructed Tan's scheme and made sure it give exact answer  for a piecewise linear field with a derivative discontinuity along the interface. 
Then I modified and verified the interpolation scheme according from the paper:
See Eqn. (27)-(28) in Sec. 3.5 in
https://arxiv.org/pdf/1812.06840
which do not seems to give the exact answer.
However if I understand the interpolation code in IBAMR correctly, I think it is not equivalent to the one Amin wrote in the paper neither equivalent to Tan's version. And I did the same tests with my understanding of what is in IBAMR. I got the exact answer with multiple orientation and interface positions. I think the one in the IBAMR use the interface condition as a given condition implicitly in the construction of correction terms which give equivalent answer from Tan's scheme if velocity field only have normal derivative jump across the interface. 
So the difference between the two schemes is:
The exisiting interpolation scheme in IBAMR would be second order accurate and equivalent to Tan's scheme if the velocity field only has normal derivative jumps. Tan's scheme would be second-order accurate with velocity field with jumps in both tangential and normal directions.

Reply:
OK --- I think that we only encounter normal derivative discontinuities, so the current scheme seems OK, but it is unfortunate that it doesn't agree with what is in the paper.

@AminKolah, do you have any time to look at this? My inclination right now is just to switch the implementation to use Tan's scheme and not worry about maintaining the original scheme (even as an option).

Reply:
@boyceg I think Qi nailed it. The difference is pretty straightforward, but I totally agree the twisted description in the paper made it a bit confusing. It looks like there's an extra normal vector factor in my formulation that needs to go. Sorry about that! Looks like we've fixed at least a couple of errors in the first IIM paper by now, but thankfully, those same mistakes didn't make it into the rigid and flexible papers (lesson learned!)."

Reply:
I'll go ahead and fix that error in the arxiv version.  There were also a couple of other things that we have already fixed in the arxiv (a sign error in the jump and the inverted surface Jacobian definition). Maybe it makes sense at this point to publish an erratum that just fix those in the Eqs?  @boyceg Let me know what you think. I have some time now to work on this if you and the rest of the co-authors agree.

Reply:
Yeah if there are errors in formulae it might be good to publish an erratum.

Reply:
@AminKolah Hello Amin, 
As I tried to formulate our new paper with small mfac. I had difficulties with grid convergence tests and reproduce the convergence rate as state in the paper. I am wondering if you could double check and reconfirm the convergence tests results in the paper with finner grid?

Reply:
Yes, that seems right. We all agree that there is deteriorated convergence for a finer grid at this point, at least when using the same form of penalty parameter as in the original work.

Reply:
I have something coming up on Monday, but I should be able to add my comments on the draft by next Friday, or just on the grid convergence if that's the only part you want me to look at? To better follow the flow of info in the manuscript, it'd be good to have all the important sections in place before reviewing. Oh, and I believe your overleaf account is not premium, so I'm unable to add edits with track changes.  That is however not a problem. I’ll download the PDF and add my comments directly to the file instead. In any case, we can continue discussing this via email.